docs/implementation/user_api/Authentication.md

```diff
@@ -0,0 +1,106 @@
+    "login_hint": "AXXXXX@alunos.isel.pt"
+}
+```
+
+For the `POLL` notification method the response is for example:
+
+```json
+{
+    "auth_req_id": "Unique ID of the Authentication Request",
+    "expires_in": 120,
+    "interval": 2
+}
+```
+
+This response follows the [Successful Authentication Request Acknowledgement](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#successful_authentication_request_acknowdlegment) of the OpenID Client Initiated Backchannel Authentication.
+
+If an error has occurred the [Authentication Error Response](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#auth_error_response) applies.
+
+After the flow #6 the server will send an email challenge to the user email (for example, `AXXXXX@alunos.isel.pt`), whilst the user needs to acknowledge this challenge by logging in to the `alunos.isel.pt` email server and accepting the challenge (a URL that redirects back to the server).
+
+If the `POLL` method was chosen the user should poll the server, with the specified interval between poll requests, and the server will respond if the authentication request has been fulfilled by the user.
+
+The responses sent by the server to the client if an error has been found, or if the user hasn't fulfilled the request follow the [OpenID Token Error Response](https://openid.net/specs/openid-connect-core-1_0.html#TokenErrorResponse).
+
+If the user has fulfilled the authorization request the response is presented as follows:
+
+```json
+{
+    "code": "The code used to obtain the access_token"
+}
+```
+
+The received code can be then used to obtain the `access_token` and more information by the client (the user gives the `code` to the client application). The response to the authorization code grant is:
+
+```json
+{
+    "id_token": "...",
+    "access_token": "...",
+    "refresh_token": "...",
+    "expires_in": 120
+}
+```
+
+Where this response follows the OpenID standard.
+
+The available error responses to the client polling event are defined in the [OAuth 2.0 Authorization Framework](https://tools.ietf.org/html/rfc6749#section-5.2), however the error codes presented in the [OAuth 2.0 Device Authorization Grant](https://tools.ietf.org/html/rfc8628#section-3.5) are also applicable as per the [OpenID CIBA Documentation](https://openid.net/specs/openid-client-initiated-backchannel-authentication-core-1_0.html#token_error_response).
+
+## Security Considerations
+
+The `scope` parameter must be validated for each request depending on the presented `client_id`.
+
+A malicious user may initiate an authentication request and therefore trick a user into accepting the request. For instance, if the request is using the email method, the user must be aware of the client that initiated the request and time, so he/she can make an informed decision about allowing or denying the request.
```

Comment:
This means the email requires enough information for the user to make a well informed decision.